packages/relay/src/lib/services/rawTxSynchronizeService/RawTxSynchronizeService.ts

```diff
@@ -0,0 +1,152 @@
+// SPDX-License-Identifier: Apache-2.0
+
+import { Mutex, withTimeout } from 'async-mutex';
+import { randomUUID } from 'crypto';
+import { LRUCache } from 'lru-cache';
+import { Logger } from 'pino';
+
+/**
+ * Represents the state of a lock for a specific sender address.
+ * Encapsulates both the mutex and the active session keys for that lock.
+ */
+interface LockState {
+  /** The mutex used for synchronization */
+  mutex: Mutex;
+  /** Set of active session keys that can release this lock */
+  activeSessionKeys: Set<string>;
+}
+
+export class RawTxSynchronizeService {
```

Comment:
Why calling it RawTxSynchronizeService, and not something general, that can be used accross the app, whit, a feature namespace for keys?